### PR TITLE
DVISA-892: (feature) Uncertainty Visibility Options

### PR DESCRIPTION
**Pull request by p1p3**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

> Feature: Add a visibility option for the uncertainties in the following tools:  Rectangle, Circle & Length.

* **What is the current behavior?** (You can also link to an open issue here)

> Uncertainties are always displayed.

* **What is the new behavior (if this is a feature change)?**

> 1. Centralise the logic to format the measurements from different tools like Rectangle, Circle & Length.
> 2. A new base tool `BaseMeasurmentTool` that includes the dynamic property `displayUncertainties` implemented as suggested in this[ cornerstoneTools issue ](https://github.com/cornerstonejs/cornerstoneTools/issues/1110#issuecomment-543735549)
> 3. Uncertainties are shown/hidden depending on the state of the displayUncertainties.
> 
> 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

> No.

### CODE REVIEW

File: src/tools/annotation/CircleRoiTool.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import external from './../../externalModules.js';
-import BaseAnnotationTool from '../base/BaseAnnotationTool.js';
+import BaseMeasurmentTool from '../base/BaseMeasurmentTool.js';
```
Comment (fabioir-dedalus) [inline src/tools/annotation/CircleRoiTool.js:2]:
```suggestion
import BaseMeasurementTool from '../base/BaseMeasurementTool.js';
```

File: src/tools/annotation/CircleRoiTool.js [hunk 2]
```diff
@@ -38,9 +39,9 @@ const logger = getLogger('tools:annotation:CircleRoiTool');
  * @memberof Tools.Annotation
  * @classdesc Tool for drawing circular regions of interest, and measuring
  * the statistics of the enclosed pixels.
- * @extends Tools.Base.BaseAnnotationTool
+ * @extends Tools.Base.BaseMeasurmentTool
```
Comment (fabioir-dedalus) [inline src/tools/annotation/CircleRoiTool.js:42]:
```suggestion
 * @extends Tools.Base.BaseMeasurementTool
```

File: src/tools/annotation/LengthTool.js [hunk 3]
```diff
@@ -20,6 +20,7 @@ import throttle from '../../util/throttle';
 import { getModule } from '../../store/index';
 import * as measurementUncertainty from '../../util/measurementUncertaintyTool.js';
 import roundToDecimal from '../../util/roundToDecimal.js';
+import { formatLenght } from '../../util/formatMeasurment';
```
Comment (fabioir-dedalus) [inline src/tools/annotation/LengthTool.js:23]:
```suggestion
import { formatLenght } from '../../util/formatMeasurement';
```

File: src/tools/annotation/LengthTool.js [hunk 4]
```diff
@@ -30,9 +31,9 @@ const logger = getLogger('tools:annotation:LengthTool');
  * @class LengthTool
  * @memberof Tools.Annotation
  * @classdesc Tool for measuring distances.
- * @extends Tools.Base.BaseAnnotationTool
+ * @extends Tools.Base.BaseMeasurmentTool
```
Comment (fabioir-dedalus) [inline src/tools/annotation/LengthTool.js:34]:
```suggestion
 * @extends Tools.Base.BaseMeasurementTool
```

File: src/tools/annotation/LengthTool.js [hunk 5]
```diff
@@ -30,9 +31,9 @@ const logger = getLogger('tools:annotation:LengthTool');
...
  */
-export default class LengthTool extends BaseAnnotationTool {
+export default class LengthTool extends BaseMeasurmentTool {
```
Comment (fabioir-dedalus) [inline src/tools/annotation/LengthTool.js:36]:
```suggestion
export default class LengthTool extends BaseMeasurementTool {
```

File: src/tools/base/BaseMeasurmentTool.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+   * @returns {void}
+   */
+  get displayUncertainties() {
+    return typeof this.options.displayUncertainties === 'undefined'
+      ? Boolean(this.configuration.displayUncertainties)
+      : Boolean(this.options.displayUncertainties);
+  }
+}
+
+export default BaseMeasurmentTool;
```
Comment (fabioir-dedalus) [inline src/tools/base/BaseMeasurmentTool.js:26]: The name of the file and class contain a typo 😅

File: src/util/formatMeasurment.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
+import { translate, localizeNumber } from './localization/localization.utils';
```
Comment (fabioir-dedalus) [inline src/util/formatMeasurment.js:1]: The file name contains a typo

File: src/util/formatMeasurment.test.js [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+          hasPixelSpacing,
+          uncertainty,
+          displayUncertainties
+        );
+
+        expect(result).toEqual(expected);
+      }
+    );
+  });
+});
```
Comment (fabioir-dedalus) [inline src/util/formatMeasurment.test.js:133]: Nice tests 😍

File: src/tools/base/BaseMeasurmentTool.js (hunk 6 above)
Comment (p1p3) [reply src/tools/base/BaseMeasurmentTool.js]: Good catch!

**Review by aminahasimbegovic — APPROVED** [review]
🚀

**Pull request merged**